qa/tasks/cephfs/filesystem.py

```diff
@@ -207,6 +207,18 @@ def hadfailover(self, status):
         #all matching
         return False
 
+    def hadfailover_rank(self, fscid, status, rank):
```

Comment:
This does an implicit `ceph fs dump` in `Filesystem.get_rank`.

I think a better place for this is in `FSStatus` with this signature:

`def had_failover_rank(self, fscid, rank, status2):`

Reply:
Looks good to me. Will update it.